src/codegen/codegen_c_visitor.cpp

```diff
@@ -1696,7 +1697,7 @@ void CodegenCVisitor::visit_eigen_linear_solver_block(ast::EigenLinearSolverBloc
 
     printer->add_newline();
     printer->add_line(
-        "X = Eigen::PartialPivLU<Eigen::Ref<Eigen::Matrix<{0}, {1}, {1}>>>(Jm).solve(F);"_format(
+        "X_operator = Eigen::PartialPivLU<Eigen::Ref<Eigen::Matrix<{0}, {1}, {1}>>>(Jm).solve(F);"_format(
```

Comment:
also, that doesn't really fix the problem, does it? What if "X_operator" is already in use? should you not make it dynamic by choosing a variable name that is not found in the symbol table? I would try "X", if found, then do "X_[four random chars].